test/e2e/e2e_test.go

```diff
@@ -1197,6 +1199,176 @@ var _ = Describe("e2e non-vxlan external gateway and update validation", func()
+			if err != nil {
+				framework.Failf("Expected container %s running on %s error %v", ovnContainer, ovnHaWorkerNode, err)
+			}
+		}
+	})
+
+	AfterEach(func() {})
+
+	It("Should validate the egress firewall policy functionality against remote hosts", func() {
+		var validSrcIP net.IP
+		srcPodName := "e2e-egress-fw-src-pod"
+		command := []string{"bash", "-c", "sleep 20000"}
+		frameworkNsFlag := fmt.Sprintf("--namespace=%s", f.Namespace.Name)
+		testContainer := fmt.Sprintf("%s-container", srcPodName)
+		testContainerFlag := fmt.Sprintf("--container=%s", testContainer)
+		// egress firewall crd yaml configuration
+		var egressFirewallConfig = fmt.Sprintf(`kind: EgressFirewall
+apiVersion: k8s.ovn.org/v1
+metadata:
+  name: default
+  namespace: %s
+spec:
+  egress:
+  - type: Allow
+    to:
+      cidrSelector: 8.8.8.8/32
+  - type: Allow
+    to:
+      cidrSelector: 1.1.1.0/24
+    ports:
+      - protocol: TCP
+        port: 80
+  - type: Deny
+    to:
+      cidrSelector: 0.0.0.0/0
+`, f.Namespace.Name)
+		// write the config to a file for application and defer the removal
+		if err := ioutil.WriteFile(egressFirewallYamlFile, []byte(egressFirewallConfig), 0644); err != nil {
+			framework.Failf("Unable to write CRD config to disk: %v", err)
+		}
+		defer func() {
+			if err := os.Remove(egressFirewallYamlFile); err != nil {
+				framework.Logf("Unable to remove the CRD config from disk: %v", err)
+			}
+		}()
+		// create the CRD config parameters
+		applyArgs := []string{
+			"apply",
+			frameworkNsFlag,
+			"-f",
+			egressFirewallYamlFile,
+		}
+		framework.Logf("Applying EgressFirewall configuration: %s ", applyArgs)
+		// apply the egress firewall configuration
+		framework.RunKubectlOrDie(applyArgs...)
+		// non-ha ci mode runs a set of kind nodes prefixed with ovn-worker
+		ciWorkerNodeSrc := ovnWorkerNode
+		if haMode {
+			// ha ci mode runs a named set of nodes with a prefix of ovn-control-plane
+			ciWorkerNodeSrc = ovnHaWorkerNode
+		}
+		nodeIP, err := runCommand("docker", "inspect", "-f", ciNetworkFlag, ciWorkerNodeSrc)
+		if err != nil {
+			framework.Failf("failed to get the node ip address from node %s %v", ciWorkerNodeSrc, err)
+		}
+		nodeIP = strings.TrimSuffix(nodeIP, "\n")
+		if ip := net.ParseIP(nodeIP); ip == nil {
+			framework.Failf("Unable to retrieve a valid address from container %s with inspect output of %s", ciWorkerNodeSrc, nodeIP)
+		}
+		framework.Logf("the pod side node is %s and the source node ip is %s", ciWorkerNodeSrc, nodeIP)
+		podCIDR, err := getNodePodCIDR(ciWorkerNodeSrc)
+		if err != nil {
+			framework.Failf("Error retrieving the pod cidr from %s %v", ciWorkerNodeSrc, err)
+		}
+		framework.Logf("the pod cidr for node %s is %s", ciWorkerNodeSrc, podCIDR)
+		// create the pod that will be used as the source for the connectivity test
+		createGenericPod(f, srcPodName, ciWorkerNodeSrc, command)
+		// wait for pod setup to return a valid address
```

Comment:
you can use this helper function, I think that there is no need to validate the IP, it's already validated in k8s
https://github.com/ovn-org/ovn-kubernetes/pull/1561/files#diff-6413e1dbafd6dd63b5d907ea3ae6f6f9R155-R171

Reply:
The need for the `wait.PollImmediate` is because framework test pods are returning from `e2epod.WaitForPodNotPending` before they have an IP and the tests will fail/flake most of the time since there isn't an IP assigned yet. More detailed description [here](https://github.com/ovn-org/ovn-kubernetes/pull/1612#discussion_r471716294) on why I changed it to look for the IP annotation. Ty!